metricsanalyzer/SAXHandler.java

```diff
@@ -0,0 +1,133 @@
+    }
+    
+    public void startElement(String uri, String localName, String qname, Attributes attr) {
+        //System.out.println("Start element: qname: " + qname);
+        length = attr.getLength();
+        String zero = "0";
+        if (qname.equals("Metric")) {
+            metricID = attr.getValue("id");
+            metricName = attr.getValue("description");
+            metric = metricID+","+metricName;
+            if (metricID.equals("NOP") || metricID.equals("TLOC")) {
+                per = "---";
+            }
+            metricsList.addElement(metric);
+           // System.out.println(metric);
+        } else if (qname.equals("Values")) {
+            per = attr.getValue("per");
+            total = attr.getValue("total");
+            max = attr.getValue("max");
+            stddev = attr.getValue("stddev");
+            avg = attr.getValue("avg");
+            values = metricID+","+revisionNum+","+per+","+total+","+avg+","+stddev+","+max;
+            systemMetrics.addElement(values);
+           // System.out.println(values);
+        } else if (qname.equals("Value")) {
+            if (per.equals("type")) {
+                typeName = attr.getValue("name");
+                typeSource = attr.getValue("source");
+                typePackage = attr.getValue("package");
+                typeValue = attr.getValue("value");
+                typeData = typeName+","+revisionNum+","+metricID+","+typeSource+","+typePackage+","+typeValue;
+                classMetrics.addElement(typeData);
+              //  System.out.println(typeData);
+            } else if (per.equals("packageFragment")) {
+                directoryName = attr.getValue("name");
+                packageName = attr.getValue("package");
+                packageValue = attr.getValue("value");
+                packageData = directoryName+","+revisionNum+","+metricID+","+packageName+","+packageValue;
+                packageMetrics.addElement(packageData);
+               // System.out.println(packageData);
+            }
+            
+            else if (per.equals("method")) {
+                methodName = attr.getValue("name");
+                methodSource = attr.getValue("source");
+                methodPackage = attr.getValue("package");
+                methodValue = attr.getValue("value");
+                methodData = methodName+","+revisionNum+","+metricID+","+methodSource+","+methodPackage+","+methodValue;
+                methodMetrics.addElement(methodData);
+               // System.out.println(methodData);
+            } else {
+                total = attr.getValue("total");
+                values = metricID+","+revisionNum+","+zero+","+total+","+zero+","+zero+","+zero;
+                systemMetrics.addElement(values);
+               // System.out.println(values);
+            }
+        }
+    }
+    
+    public void endElement(String uri, String localName, String qname) {
+        //System.out.println("End element: qname: " + qname);

```

Comment:
I'd classify it as a unnecessary since we can't actually ask the author what it is intended to do. @kelahm